### ISSUE TITLE
RevEng: Don't make navigation properties virtual

### ISSUE THREAD

**Issue by bricelam** [aspnet/EntityFramework#6326]
Reverse engineer generates `virtual` collection and reference navigation properties.

``` C#
class Post {
    public virtual Blog Blog { get; set; }
    public virtual ICollection<Comment> Comments { get; set; }
}
```

Given that we don't generate proxy types, should these be virtual?

**Comment by divega** [comment]
**Triage:** we shouldn't be generating virtual properties by default.

**Comment by gdoron** [comment]
When you'll add Lazy load, will we need to change everything to `virtual`?

**Comment by divega** [comment]
@gdoron It depends on how lazy loading ends up being implemented. If we did it by generating proxy types at runtime then yes, very likely virtual would be needed to override the property getter. But there are other ways lazy loading could be implemented that wouldn't require it. 

It is possible that we will revisit this decision once we know how lazy loading will look like, but when we discussed this today we thought that making the change for 1.1 help us reduce the impact of changing the behavior later, e.g. if at some point we decided to lazy load by default, it won't affect as much existing code.
